pslogin/src/main/scala/LoginSessionActor.scala

```diff
@@ -25,9 +37,27 @@ class LoginSessionActor extends Actor with MDCContextAware {
   var sessionId : Long = 0
   var leftRef : ActorRef = ActorRef.noSender
   var rightRef : ActorRef = ActorRef.noSender
+  var accountIntermediary : ActorRef = Actor.noSender
 
   var updateServerListTask : Cancellable = DefaultCancellable.obj
 
+  var ipAddress : String = ""
+  var hostName : String = ""
+  var canonicalHostName : String = ""
+  var port : Int = 0
+
+  val serverName = WorldConfig.Get[String]("worldserver.ServerName");
+
+  // This MUST be an IP address. The client DOES NOT do name resolution 
+  var serverHost : String = if (WorldConfig.Get[String]("worldserver.Hostname") != "")
+    InetAddress.getByName(WorldConfig.Get[String]("worldserver.Hostname")).getHostAddress
+  else
+    LoginConfig.serverIpAddress.getHostAddress
+
+  val serverAddress = new InetSocketAddress(serverHost, WorldConfig.Get[Int]("worldserver.ListeningPort"))
+
+  private val numBcryptPasses = 4
```

Comment:
Isn't this way, way too low to avoid brute forcing by modern CPUs/GPUs?

Reply:
Yes, it is. 10 would be better. https://stackoverflow.com/a/50470009 I was concerned with increasing it and it breaking the existing passwords, but looking at it more, it looks like it stores the number of rounds in the hash itself. I can increase it.

Reply:
> 
> 
> Isn't this way, way too low to avoid brute forcing by modern CPUs/GPUs?

That brute force is non only if you have a copy ? Because with a DB not over internet + we can limite the number of try on login (3 login in 1 min => block), I'm not sure that's needed. No ? (I'm not a security expert)

Reply:
The primary use is to slow down hash cracking in the event that the database is compromised. MD5 for example, can be run billions of times a second allowing for any passwords hashed with it to be cracked in a matter of seconds with GPUs. With a work-factor of 10 for bcrypt, if our DB is leaked, it will take ~100ms for each hash query. 10h/s is much better than 1000000000h/s ;)

Reply:
Ok, in case of compromised :)